ext/standard/php_lcg.h

```diff
@@ -1,38 +1 @@
-/*
-   +----------------------------------------------------------------------+
-   | Copyright (c) The PHP Group                                          |
-   +----------------------------------------------------------------------+
-   | This source file is subject to version 3.01 of the PHP license,      |
-   | that is bundled with this package in the file LICENSE, and is        |
-   | available through the world-wide-web at the following url:           |
-   | https://www.php.net/license/3_01.txt                                 |
-   | If you did not receive a copy of the PHP license and are unable to   |
-   | obtain it through the world-wide-web, please send a note to          |
-   | license@php.net so we can mail you a copy immediately.               |
-   +----------------------------------------------------------------------+
-   | Author: Sascha Schumann <sascha@schumann.cx>                         |
-   +----------------------------------------------------------------------+
-*/
-
-#ifndef PHP_LCG_H
-#define PHP_LCG_H
-
-#include "ext/standard/basic_functions.h"
-
-typedef struct {
-	int32_t s1;
-	int32_t s2;
-	int seeded;
-} php_lcg_globals;
-
-PHPAPI double php_combined_lcg(void);
-
-PHP_MINIT_FUNCTION(lcg);
-
-#ifdef ZTS
-#define LCG(v) ZEND_TSRMG(lcg_globals_id, php_lcg_globals *, v)
-#else
-#define LCG(v) (lcg_globals.v)
-#endif
-
-#endif
+#include "ext/random/php_random.h"
```

Comment:
Wouldn't be better to remove this file and similar ones below? I mean this would be going to new minor version which means that we don't need to keep compatibility for extensions so if anything includes this, it should be changed.

Reply:
I am doing this because in a previous Internals thread there was discussion that compatibility should be maintained.
I would like to remove this as well, but perhaps the content of the RFC makes it impossible to do so at this time.
I will add it to the task list of RNG improvements in PHP 8.3.

Reply:
I don't really understand what would be the point of waiting till 8.3? It's the same if one does the change in 8.2 or 8.3. Also how many extensions actually include those headers? I guess it will be really small number anyway.

Reply:
I suggested that this header file be removed too but in the course of the RFC discussion, some suggested that it would be better to do it this way.

https://externals.io/message/115918#116139

Perhaps, if we remove this header, the RFC **Backward Incompatibile Changes** should state that the old header file will be removed. Unfortunately, the RFCs is already closed, and it is difficult to change them now.

If we can get more people to agree, we could make the change, but I would like to hear from core members as I do not have the knowledge surrounding exceptional changes.

Reply:
Ok leave it as it is - it's not such a big deal and we can remove it later.

Reply:
For the reference: The follow-up is at #9052.